### PR TITLE
Tile drawing implementation and bug fixes

### PR DESCRIPTION
**Pull request by white-axe**
This pull request re-adds the ability to change tiles on the map.

I also snuck in some fixes related to the map and tilepicker.

### CODE REVIEW

**Comment by white-axe** [comment]
Checks seem to be broken

**Comment by Speak2Erase** [comment]
Yeah, GitHub keeps terminating the actions for some reason

It's probably because of resource usage at link time? I tried swapping to old and mold, but it didn't help

**Comment by Speak2Erase** [comment]
I think I'll look into self hosted runners for Autobuild

File: src/tabs/map.rs [hunk 1]
```diff
@@ -57,6 +57,156 @@ impl Tab {
             force_close: false,
         })
     }
+
+    fn recompute_autotile(&self, map: &rpg::Map, position: (usize, usize, usize)) -> i16 {
```
Comment (Speak2Erase) [inline src/tabs/map.rs:61]: Woah, how did you figure this out?

Comment (white-axe) [reply src/tabs/map.rs:61]: I looked at how RPG Maker XP behaves when you place autotiles and used how the autotile variants are ordered numerically to deduce this algorithm

**Review by Speak2Erase — APPROVED** [review]

**Pull request merged**